Add sass mixins and partials

bholaday opened the pull request:
**Mixins**
- _backgrounds - create a repeating striped background with CSS
    - syntax - `@include striped-bg($color1, $color2, $style)`
        - output - `background: repeating-linear-gradient($degrees, $color1, $color1 $line1end, $color2 $line1end, $color2 $line2end);`

- _calculations - convert px to rem or em
    - syntax - `@include calc-toRem($n)`
        - output - `##rem`
    - syntax - `@include calc-toEm($n)`
        - output - `##em`
    - syntax - `@include calc-ratioLgMd`
        - output - `##`

- _fonts - set variables for font-stacks

**Partials**
- _button - create default styles for elements that use the `button` class

bholaday commented on components/css/mixins/_fonts.scss:
```diff
@@ -0,0 +1,10 @@
+$font-champion-welter:"Champion Welterwt A","Champion Welterwt B",$font-archivo-black;
+$font-champion-middle:"Champion Middlewt A","Champion Middlewt B",$font-archivo-black;
+$font-champion-heavy:"Champion Heavywt A","Champion Heavywt B",$font-archivo-black;
+$font-fontAwesome:FontAwesome;
```
Should this really be in the `mixins` folder or should we create a `variables` folder?

jacobvenable replied:
I would say yes because mixins should contain any SCSS that doesn't directly output CSS. So mixins, variables, etc. If we end up using ITCSS, then we'll have to move it anyways :P.

jacobvenable replied:
Perhaps before we merge, we should update each use of these fonts in the individual page SCSS files. So anywhere that uses "Arial" will be replaced by the arial font-family variable.

bholaday commented:
Should I also include the updated sass imports to actually have access to these mixins for page level sass files?

jacobvenable commented on components/css/mixins/_fonts.scss:
```diff
@@ -0,0 +1,10 @@
+$font-arial:Arial,"Helvetica Neue",Helvetica,sans-serif;
+$font-archivo-narrow:"Archivo Narrow","Arial Narrow",Arial,sans-serif;
+$font-archivo-black:'archivo_blackregular',"Arial Black","Arial Bold",Gadget,sans-serif;
```
There is a bit of a naming difference between Archivo Narrow and Archivo Black. The reason is that Archivo Narrow is currently using a font hosted on Google Fonts while Archivo Black is using a font hosted on the server. Lets go ahead and replace Archivo Black's variable value with:
`'Archivo Black',"Arial Black","Arial Bold",Gadget, sans-serif;`
We'll also start loading this font via Google Fonts instead.

bholaday commented:
I've committed some more changes. 

- updated the Arial font stack
- cleaned up the font stacks a bit
- updated the font-family declarations of sass files. I only looked for instances of Arial, so there may be other opportunities out there to reference other font variables.

NOTE: Some css font-stacks had Helvetica before Arial. The `_font.scss` declares Helvetica _**after**_ Arial, so my change will override those declarations since I've replaced them. In all situations Helvetica was a fallback for Archivo, so I think this is acceptable.

bholaday commented:
@jacobvenable Your changes look good to me.
